heidrun/nara_json.rb

```diff
@@ -0,0 +1,575 @@
+#
+#  OR
+#
+#  <parentFileUnit>
+#  <title>VALUE1</title>
+#  <parentSeries>
+#    <title>VALUE2</title>
+#    <parentCollection>
+#      <title>VALUE3</title>
+#    </parentCollection>
+#  </parentSeries>
+#  </parentFileUnit>
+#  [should be combed as VALUE3""; ""VALUE2""; ""VALUE1]"
+#
+# @param parent_file_unit [Krikri::JsonParser::Value]
+# @return [String]
+def make_relation(parent_file_unit)
+  node = parent_file_unit.node
+  title = node['title']
+  ps = node['parentSeries']
+  parent_srs_title = ps['title']
+  group_or_coll = ps.fetch('parentRecordGroup', ps['parentCollection'])
+  group_coll_title = group_or_coll['title']
+  "#{group_coll_title}; #{parent_srs_title}; #{title}"
+end
+
+# Return a string for sourceResource.description
+#
+# <generalNoteArray>
+# <generalNote>
+# <note>[VALUE]</note>
+# </generalNote>
+# </generalNoteArray>
+#
+# <scopeAndContentNote>[VALUE]</scopeAndContentNote>
+#
+# @param element [Krikri::JsonParser::Value]
+# @return [String]
+def make_description(element)
+  node = element.node
+  (node.is_a? String) ? node : node['note']
+end
+
+# <useRestriction>
+#   <note>VALUE1</note>
+#   <specificUseRestrictionArray>
+#     <specificUseRestriction>
+#       <termName xmlns=""http://description.das.nara.gov/"">VALUE2</termName>
+#     </specificUseRestriction>
+#   </specificUseRestrictionArray>
+#   <status>
+#     <termName xmlns=""http://description.das.nara.gov/"">VALUE3</termName>
+#   </status>
+# </useRestriction>
+#
+# These should be combined as "VALUE2: VALUE1 VALUE3"
+#
+# @param element [Krikri::JsonParser::Value]
+# @return [String]
+def make_rights(use_restriction)
+  node = use_restriction.node
+  note = node.fetch('note', nil)
+  sura = node.fetch('specificUseRestrictionArray', nil)
+  status = node.fetch('status', nil)
+  l_and_r_parts = \
+    [specific_rights_part(sura), genl_rights_part(note, status)].compact
+  l_and_r_parts.join ': '
+end
+
+# @see #make_rights
+def specific_rights_part(specific_use_restriction_array)
+  return nil if specific_use_restriction_array.nil?
+  sur = specific_use_restriction_array['specificUseRestriction']
+  sur = [sur] unless sur.is_a? Array
+  terms = sur.map { |el| el['termName'] }
+  terms.join ', '
+end
+
+# @see #make_rights
+def genl_rights_part(note, status)
+  [note, status['termName']].compact.join ' '
+end
+
+
+# @see #make_date_provided_label
+# @see #make_begin_date
+# @see #make_end_date
+#
+# @param node [Hash]
+# @return [String]
+def date_string(node)
+  return "" if node.nil?
+  ymd = [
+    node.fetch('year', nil), node.fetch('month', nil), node.fetch('day', nil)
+  ].compact.map { |e| "%02d" % e }.join '-'
+  qualifier_node = node.fetch('dateQualifier', false)
+
+  return ymd unless qualifier_node
+
+  qualifier = qualifier_node['termName']
+  (qualifier == '?') ? "#{ymd}#{qualifier}" : "#{qualifier} #{ymd}"
+end
+
+# Date and temporal fields
+#
+# FIXME:
+#
+# This may be wrong.  The original comment in this file was:
```

Comment:
They are not temporal dates. As @anarchivist says, in EAD they are similar to bulk dates or inclusive dates...i.e., just a date range